core/src/main/java/gyro/core/reference/ReferenceResolver.java

```diff
@@ -16,12 +16,119 @@
 
 package gyro.core.reference;
 
+import java.util.ArrayList;
 import java.util.List;
 
+import gyro.core.GyroException;
+import gyro.core.scope.RootScope;
 import gyro.core.scope.Scope;
+import gyro.lang.Locatable;
+import gyro.lang.ast.Node;
+import gyro.lang.ast.value.ReferenceNode;
+import gyro.lang.ast.value.ReferenceOption;
 
 public abstract class ReferenceResolver {
 
-    public abstract Object resolve(Scope scope, List<Object> arguments) throws Exception;
+    public abstract Object resolve(ReferenceNode node, Scope scope, List<Object> arguments) throws Exception;
+
+    private static ReferenceOption getOption(ReferenceNode node, String name) {
```

Comment:
These are copied from `DirectiveProcessor`. Let's find a way to reuse them rather than copy them.

Reply:
Refactored code to use a common class and a common processor to handle arguments and options